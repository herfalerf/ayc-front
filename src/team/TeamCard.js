import React from "react";
import "./TeamCard.css";

// Show information about a team member
//
// Is rendered by TeamList to show a "card" for each team member.
//
// TeamList -> TeamCard
//

function TeamCard({ name, bio, img }) {
  console.debug("TeamCard", img);

  return (
    <div>
      {img && <img src={img} alt={name} />}
      <h6>{name}</h6>
      <p>{bio}</p>
    </div>
  );
}

export default TeamCard;
