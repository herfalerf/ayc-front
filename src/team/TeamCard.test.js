import React from "react";
import { render } from "@testing-library/react";
import TeamCard from "./TeamCard";
import { MemoryRouter } from "react-router";

it("renders without crashing", function () {
  render(<TeamCard />);
});

it("matches snapshot", function () {
  const { asFragment } = render(
    <MemoryRouter>
      <TeamCard name="Test Name" bio="Test bio" img="img.jpg" />
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
});
