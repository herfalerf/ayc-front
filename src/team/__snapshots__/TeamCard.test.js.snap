// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="MuiPaper-root MuiCard-root MuiPaper-elevation1 MuiPaper-rounded"
    >
      <div
        class="MuiCardMedia-root makeStyles-image-2"
        style="background-image: url(img.jpg);"
        title="Test Name"
      />
      <div
        class="MuiCardHeader-root"
      >
        <div
          class="MuiCardHeader-content"
        >
          <span
            class="MuiTypography-root MuiCardHeader-title MuiTypography-h5 MuiTypography-displayBlock"
          >
            Test Name
          </span>
        </div>
      </div>
      <div
        class="MuiCardContent-root"
      >
        <p
          class="MuiTypography-root MuiTypography-body2"
        >
          Test bio
        </p>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
